tests/MODULE.bazel

```diff
@@ -1,6 +1,156 @@
+        "https://mirror.bazel.build/repo1.maven.org/maven2",
+    ],
+)
+
+# The compiler bridge has a dependency on compiler-interface, which has a dependency on the Scala 2
+# library. We need to set this to neverlink = True to avoid this the Scala 2 library being pulled
+# onto the Scala 3, and other Scala versions like 2.12, compiler classpath during runtime.
+annex_test.artifact(
+    name = "annex_test",
+    artifact = "scala3-sbt-bridge",
+    group = "org.scala-lang",
+    neverlink = True,
+    version = scala_3_version,
+)
+
+# The compiler bridge has a dependency on compiler-interface, which has a dependency on the Scala 2
+# library. We need to set this to neverlink = True to avoid this the Scala 2 library being pulled
+# onto the Scala 3, and other Scala versions like 2.12, compiler classpath during runtime.
+annex_test.artifact(
+    name = "annex_test",
+    artifact = "compiler-bridge_2.13",
+    group = "org.scala-sbt",
+    neverlink = True,
+    version = zinc_version,
+)
+use_repo(annex_test, "annex_test")
+
+annex_test_2_12 = use_extension("@rules_jvm_external//:extensions.bzl", "maven")
+annex_test_2_12.install(
+    name = "annex_test_2_12",
+    artifacts = [
+        "org.scala-lang:scala-compiler:{}".format(scala_2_12_version),
+        "org.scala-lang:scala-library:{}".format(scala_2_12_version),
+        "org.scala-lang:scala-reflect:{}".format(scala_2_12_version),
+        "org.specs2:specs2-common_2.12:{}".format(specs2_version),
+        "org.specs2:specs2-core_2.12:{}".format(specs2_version),
+        "org.specs2:specs2-matcher_2.12:{}".format(specs2_version),
+        "org.scalatest:scalatest_2.12:{}".format(scalatest_version),
+        "org.scalacheck:scalacheck_2.12:{}".format(scalacheck_version),
+    ],
+    fetch_sources = True,
+    lock_file = "//:annex_test_2_12_install.json",
+    repositories = [
+        "https://repo.maven.apache.org/maven2",
+        "https://maven-central.storage-download.googleapis.com/maven2",
+        "https://mirror.bazel.build/repo1.maven.org/maven2",
+    ],
+)
+
+# The compiler bridge has a dependency on compiler-interface, which has a dependency on the Scala 2
+# library. We need to set this to neverlink = True to avoid this the Scala 2 library being pulled
+# onto the Scala 3, and other Scala versions like 2.12, compiler classpath during runtime.
+annex_test_2_12.artifact(
+    name = "annex_test_2_12",
+    artifact = "compiler-bridge_2.12",
+    group = "org.scala-sbt",
+    neverlink = True,
+    version = zinc_version,
+)
+use_repo(annex_test_2_12, "annex_test_2_12")
+
+http_file = use_repo_rule("@bazel_tools//tools/build_defs/repo:http.bzl", "http_file")
+
+http_file(
+    name = "shapeless_jar",
+    downloaded_file_path = "shapeless.jar",
+    sha256 = "072c83eca9996aed92310dc7225cfc313edc7b74a3a96e2bf25459ebfc04ac96",
+    urls = ["https://repo.maven.apache.org/maven2/com/chuusai/shapeless_2.13/2.3.7/shapeless_2.13-2.3.7.jar"],
+)
+
+http_file(
```

Comment:
Similar thought here.  Can this be downloaded with rules_jvm_external instead?

Reply:
I considered migrating it to `rules_jvm_external`, but thought it was best if it continued to be fetched via `http_file`. That way, it wouldn't have `JavaInfo` or any extra providers added, which tests that `scala_import` doesn't rely on any of this information.

Reply:
Ah. Do you want to add a comment to these `http_file`s explaining why they aren't part of `rules_jvm_external`? I think others may have the same thought as me in the future.

Reply:
Will do.